src/mission_ui.cpp

```diff
@@ -17,239 +17,299 @@
 #include "translations.h"
 #include "ui.h"
 #include "ui_manager.h"
+#include "cata_imgui.h"
+#include "imgui/imgui.h"
 
-void game::list_missions()
+enum class mission_ui_tab_enum : int {
+    ACTIVE = 0,
+    COMPLETED,
+    FAILED,
+    num_tabs
+};
+
+static mission_ui_tab_enum &operator++( mission_ui_tab_enum &c )
+{
+    c = static_cast<mission_ui_tab_enum>( static_cast<int>( c ) + 1 );
+    if( c == mission_ui_tab_enum::num_tabs ) {
+        c = static_cast<mission_ui_tab_enum>( 0 );
+    }
+    return c;
+}
+
+static mission_ui_tab_enum &operator--( mission_ui_tab_enum &c )
+{
+    if( c == static_cast<mission_ui_tab_enum>( 0 ) ) {
+        c = mission_ui_tab_enum::num_tabs;
+    }
+    c = static_cast<mission_ui_tab_enum>( static_cast<int>( c ) - 1 );
+    return c;
+}
+
+class mission_ui;
+
+class mission_ui
+{
+        friend class mission_ui_impl;
+    public:
+        void draw_mission_ui();
+};
+
+class mission_ui_impl : public cataimgui::window
+{
+    public:
+        std::string last_action;
+        explicit mission_ui_impl() : cataimgui::window( _( "Your missions" ),
+                    ImGuiWindowFlags_NoResize | ImGuiWindowFlags_NoMove | ImGuiWindowFlags_NoNav |
+                    ImGuiWindowFlags_NoScrollbar | ImGuiWindowFlags_NoScrollWithMouse ) {
+        }
+
+    private:
+        void draw_mission_names( std::vector<mission *> missions, int &selected_mission,
+                                 bool &need_adjust ) const;
+        void draw_selected_description( std::vector<mission *> missions, int &selected_mission );
+
+        mission_ui_tab_enum selected_tab = mission_ui_tab_enum::ACTIVE;
+        mission_ui_tab_enum switch_tab = mission_ui_tab_enum::num_tabs;
+
+        size_t window_width = str_width_to_pixels( TERMX );
+        size_t window_height = str_height_to_pixels( TERMY );
```

Comment:
```suggestion
        size_t window_width = str_width_to_pixels( TERMX ) / 2;
        size_t window_height = str_height_to_pixels( TERMY ) / 2;
```

Full screen is a predictable size, but rather overwhelming. And missions have so little text to display at the moment. Might I suggest a window one quarter the size instead?